src/flb_utils.c

```diff
@@ -740,11 +741,24 @@ int flb_utils_write_str(char *buf, int *off, size_t size,
             if (i + hex_bytes > str_len) {
                 break; /* skip truncated UTF-8 */
             }
+
+            is_valid = FLB_TRUE;
             for (b = 0; b < hex_bytes; b++) {
+                /* Trailing characters must start with bit 1 */
+                if (b > 0 && !(str[i+b] & 0x80)) {
+                    /* Invalid unicode character. skip */
+                    flb_warn("[pack] invalid UTF-8 bytes found, skipping bytes");
```

Comment:
This is to mirror the functionality of the char point unicode shown here: https://github.com/fluent/fluent-bit/pull/4297/files#diff-d7e2a365ad467f9c9eb8283ff56105fa3a08e7fb55690a61b7137c7704a95263L722

Another solution would be to replace invalid utf-8 with the unicode replacement character: 0xEF 0xBF 0xBD �

Reply:
May be a good idea to have a different error message here. Weighing consistency vs uniqueness.

Reply:
I think you definitely want a different error message. Consistency in form and style is important, but the problem with exact duplicate error messages is that when a user reports they are seeing them, we have no way of knowing which part of the code caused it. So unique messages are actually important.

Reply:
New warning: [pack] unexpected UTF-8 continuation byte, omitting a character